pretyping/vnorm.ml

```diff
@@ -416,7 +416,7 @@ let cbv_vm env sigma c t  =
   (* This evar-normalizes terms beforehand *)
   let c = EConstr.to_constr ~abort_on_undefined_evars:false sigma c in
   let t = EConstr.to_constr ~abort_on_undefined_evars:false sigma t in
-  let v = Vmsymtable.val_of_constr env c in
+  let v = Vmsymtable.val_of_constr env (Evd.existential_opt_value0 sigma) c in
```

Comment:
Is the `to_constr` above now necessary?

Reply:
Yes and no. Yes, the term still needs to be normalized. No, `EConstr.Unsafe.to_constr` could be used instead.

So, it is mostly a matter of whether we want to encourage the use of `Unsafe` when it does not improve performances.

Reply:
I think we prefer `Unsafe` to `abort_on_undefined_evars:false` here.
Indeed the reason we have to pass `opt_value` is that we do `Unsafe` things, just hidden because the `env` type is shared between the kernel and the econstr interfaces.

Reply:
Good point. Done.

Reply:
So, this breaks things:
```coq
Definition barvm (x := true) := Eval vm_compute in x.
(* Error: The type int must be registered before this construction can be typechecked. *)
```

Any suggestion or should I just go back to using `abort_on_undefined_evars:false`?

Reply:
~~~coq
Definition barvm (t:=_) (x := true : t) := Eval vm_compute in x.
~~~
gets the same error even wth abort_on_undefined_evars (both master and current PR with Unsafe replaced), so IMO we should just fix that bug.
The issue seems to be `construct_of_constr`.

Reply:
Right, I was just reading your comments on #8059. You just saved me the hassle of exhibiting this test case. Thanks.

Reply:
Please double check that I did it correctly. I mindlessly ported your changes from #8059. I do not understand them.

Reply:
Looks ok